src/components/CGroupTable.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+  <div class="px-4 pt-2">
+    <div class="flex float-right w-48 mb-10 search relative mr-6 my-2">
+      <div class="loader absolute right-0" v-if="isLoading">Loading...</div>
+    </div>
+    <table class="w-full bg-white rounded mb-4">
+      <thead>
+        <tr class="bg-gray-100 h-8">
+          <c-th
+            v-for="(column, index) in columns"
+            :label="column.label"
+            :name="column.name"
+            :key="index"
+          ></c-th>
+        </tr>
+      </thead>
+      <tbody>
+        <template v-for="g in groups">
+          <c-group-header
+            :key="g.groupId"
+            :groupId="g.groupId"
+            :messages="g.messages"
+            @onToggle="toggleRow"
+          ></c-group-header>
+          <c-group-content
+            v-if="openedRows.includes(g.groupId)"
+            :key="g.groupId + '0'"
```

Comment:
why this '0' ?

Reply:
allows to avoid the collision among keys, it is possible to add something more interpretable like {header0, header1,...} and {content0, content1..}, however, the efficiency will decrease.

Reply:
Basically you should never choose performance over comprehension of the code until performance is a problem. I don't think there is any performance issue here ? so use more interpretable name.